src/Modules/TeamInfoModule/Controllers/TeamInfoEventController.cs

```diff
@@ -36,14 +37,27 @@ public async Task OnScoresAsync(object sender, ScoresEventArgs eventArgs)
             await teamInfoService.SetModeIsTeamsAsync(true);
         }
 
-        var teamInfos = eventArgs.Teams.ToList();
-        var team1Points = teamInfos[0]!.MatchPoints;
-        var team2Points = teamInfos[1]!.MatchPoints;
+        if (teamInfoService.ShouldUpdateTeamPoints(eventArgs.Section))
+        {
+            var teamInfos = eventArgs.Teams.ToList();
+
+            await teamInfoService.UpdatePointsAsync(
+                teamInfos[0]!.MatchPoints,
+                teamInfos[1]!.MatchPoints,
+                teamInfoService.ShouldExecuteManiaScript(eventArgs.Section)
+            );
```

Comment:
```suggestion
            await teamInfoService.UpdatePointsAsync(
                eventArgs.Teams.FirstOrDefault()?.MatchPoints ?? 0,
                eventArgs.Teams.Skip(1).FirstOrDefault()?.MatchPoints ?? 0,
                teamInfoService.ShouldExecuteManiaScript(eventArgs.Section)
            );
```

You should be able to use standard collection methods to simplify these, as well as adding a default value instead of just ignoring the possible null.